llvm/lib/Transforms/Scalar/NaryReassociate.cpp

```diff
@@ -565,14 +565,24 @@ NaryReassociatePass::findClosestMatchingDominator(const SCEV *CandidateExpr,
   // optimization makes the algorithm O(n).
   while (!Candidates.empty()) {
     // Candidates stores WeakTrackingVHs, so a candidate can be nullptr if it's
-    // removed
-    // during rewriting.
-    if (Value *Candidate = Candidates.back()) {
+    // removed during rewriting.
+    if (Value *Candidate = Candidates.pop_back_val()) {
```

Comment:
This change seems problematic. Previously, we returned a candidate without pop_back the stack. With this change, the stack popped before returning the candidate. I have seen some problems.